docs/spec.md

```diff
@@ -2,4 +2,253 @@
+
+* `remainingAttempts`: number of attempts remaining before the SHL is disabled
+
+### Limitations on Manifest `.files.location` links
+
+The Data Sharer SHALL ensure that `.files.location` links can be dereferenced
+without additional authentication, and that they are short-lived. The lifetime
+of `.files.location` links SHALL NOT exceed one hour. The Data Sharer MAY create
+one-time-use `.files.location` links that are consumed as soon as they are
+dereferenced.
+
+The Data Recipient SHALL treat any manifest file locations as short-lived and
+potentially limited to one-time use. The Data Recipient SHALL NOT attempt to
+dereference a manifest's `.files.location` link more than one hour after
+requesting the manifest, and SHALL be capable of re-fetching the manifest to
+obtain fresh `location` links in the event that they have expired or been
+consumed.
+
+
+The following optional step may occur sometime after a SHLink is shared:
+
+### Optional: Periodically Re-fetch Updated Files 
+When the original QR includes the `L` flag for long-term use, the Data Recipient
+MAY re-fetch contents at any time. When re-fetching data, the Data Recipient
+SHALL begin from the final redirect URL it discovered when first following the
+QR Link, because the first-stage redirection may no longer be in place. To
+support long-term access, the Manifest HTTP response SHOULD include an
+[`Expires` header](https://www.imperva.com/learn/performance/cache-control)
+indicating the time when a subsequent fetch should be attempted.
+
+---
+
+## SHLink Manifest File Format
+
+The SHLink Manifest File is a JSON file with a `files` array where each entry includes:
+
+* `contentType`: One of  the following values:
+    * `"application/smart-health-card"` or
+    *  `"application/smart-api-access"` or 
+    *  `"application/fhir+json"`
+* `location` (SHALL be present if no `embedded` content is included):
+URL to the file.  This URL SHALL be short-lived and intended for single use. For
+example, it could be a short-lifetime signed URL to a file hosted in a cloud
+storage service.
+* `embedded` (SHALL be present if no `location` is included):
+JSON string directly embedding the encrypted contents of the file as a compact
+JSON Web Encryption string (see ["Encrypting"](#encrypting-and-decrypting-files)).
+If present, the `embedded` value SHALL be up-to-date as of the time the
+manifest is requested.
+
+##### Example SHLink Manifest File
+
+```json
+{
+  "files": [{
+    "contentType": "application/smart-health-card",
+    "location": "https://bucket.cloud.example.org/file1?sas=MFXK6jL3oL3SI_lRfi_-cEfzIs5oHs6rRWmrsCAFzvk"
+  }, 
+  {
+    "contentType": "application/smart-health-card",
+    "embedded": "eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..8zH0NmUXGwMOqEya.xdGRpgyvE9vNoKzHlr4itKKW2vo<snipped>"
```

Comment:
I understand adding `contentType` for `embedded` data, but why add `contentType` if a `location` is specified?

Reply:
> why add contentType

Aim was to help clients avoid fetching data they can't handle. Like if you don't deal with `application/smart-access-token`, you can avoid fetching that.

Reply:
Ah got it, helpful.